src/classes/HiFiMixerSession.ts

```diff
@@ -639,12 +653,15 @@ export class HiFiMixerSession {
             // and pass along the error message. `this._onConnectionStateChange`
             // will disconnect and clean up for us.
             this._onConnectionStateChange(HiFiConnectionStates.Failed, { success: false, error: error });
+            return false;
         })
         .finally(() => {
             this._raviSignalingConnection.removeStateChangeHandler(tempUnavailableStateHandler);
             this.concurrency = 0;
             this._tryingToConnect = false;
         });
+
+        return true;
```

Comment:
I guess the promise itself is evaluated for effect, and the value not used anywhere? 
So it's intended to return true, and not the promise?

Reply:
And the doc string says it returns a boolean indicating success or failure, but we don't actually know the success at this point in time, right?

Reply:
This is why I needed more eyes on it! :-) Thank you, @howard-stearns .  I've moved the `return true` inside the last `then`, and at that point we SHOULD know that it was successful. So I believe that with this _latest_ change, it is appropriate for `HiFiMixerSession.connectToHiFiMixer()` to be considered synchronous, and the boolean it returns should, in fact, indicate success or failure. (I think.) Does that sound right?

Reply:
@howard-stearns Thank you again so much for your help with this! `connectToHiFiMixer()` now returns `void` with an explicit note about getting the "return value" ONLY through the use of callbacks. The test has been updated to use a callback and check to make sure it gets called with the appropriate values. Whew.